### PR TITLE
Add missing fqn -> storageIdent conversion

### PR DESCRIPTION
**Pull request by jeeminso**
Stumbled on this which should be left out from one of drop-column PRs.

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
Thanks for catching!

I think we need a test here. It was not caught by `testNeqAnyOnArrayLiteral` because on `LuceneQueryBuilderTest`-s we don't assign OID-s and thus `probe.storageIdent()` ==  `probe.column().fqn()`. 

Can we have a `LuceneQueryBuilderIntegrationTest` case for that?

File: server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: Thanks for catching!

I think we need a test here. It was not caught by testNeqAnyOnArrayLiteral because on LuceneQueryBuilderTest-s we don't assign OID-s and thus probe.storageIdent() ==  probe.column().fqn().

Can we have a LuceneQueryBuilderIntegrationTest case for that?

Comment (mfussenegger) [reply server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: I don't think we want to add additional integration tests for all the query variants either.

We could have additional unit tests that test the query matching using the `QueryTester` without asserting the query output. Similar to `test_is_null_on_columns_without_doc_values`  (minus the _field_names stuff)

Comment (jeeminso) [reply server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: As Baur said, with unit tests, I think we cannot determine whether we used `ref.storageIdent()` or `ref.column().fqn()` to generate Lucene queries since both return the column name (not oid).
For example, if we intentionally change https://github.com/crate/crate/blob/bd44a8890c7754d0159458065b36d8bad93983d5/server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java#L119 back to `ref.column().fqn()`, `test_is_null_on_columns_without_doc_values` will still pass.

Comment (seut) [reply server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: We could add a unit test variant which uses a `QueryTester` instance that uses a real OID supplier rather than the default one defined here https://github.com/crate/crate/blob/master/server/src/testFixtures/java/io/crate/testing/QueryTester.java#L89.

Comment (jeeminso) [reply server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: Thank you, added a test but just to this specific case. I will expand to cover other `toQuery` implementations.

Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: >I will expand to cover other toQuery implementations.

We probably need to re-use `queryTester`. 
Add API to set and pass through to the `queryTester`, reset after each test to unassigned.

Also, it looks like we always use single column in tests, ie supplier returning same value might be okay. 

However, we might consider having a supplier with actual increment (all columns in `LuceneQueryBuilderTest.prepare` will get different oids) and adjust tests accordingly - ie update assertions to use different oids across tests cases and not always "1"

Comment (jeeminso) [reply server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java:71]: > However, we might consider having a supplier with actual increment (all columns in LuceneQueryBuilderTest.prepare will get different oids) and adjust tests accordingly - ie update assertions to use different oids across tests cases. Just to be safe and be able to use 2+ columns in the same test.

Thanks I went this direction to let all tests to take effect and actually found another missing conversion from `NumTermsPerDocQuery`.

**Comment by jeeminso** [comment]
Since all query tests now use oids, no need to add new tests.

File: server/src/testFixtures/java/io/crate/testing/QueryTester.java
Comment (seut) [inline server/src/testFixtures/java/io/crate/testing/QueryTester.java:92]: The whole intention of not using a real OID supplier was that we can still assert in all query tests for real column names. Using OID's will decrease readability completely, it would be very hard to see if the expected outcomes are valid. What about @mfussenegger suggestion to add test variants for the concrete issues which do not assert the query output? I highly suggest to move into such a direction instead.

**Review by seut — CHANGES_REQUESTED** [review]

File: server/src/testFixtures/java/io/crate/testing/QueryTester.java
Comment (seut) [reply server/src/testFixtures/java/io/crate/testing/QueryTester.java:92]: Also, I made the `ColumnOidSupplier` private by intent, to not use it in such tests, but instead, if really needed, use a custom implementation (its `LongSupplier` so a custom impl is very simple).

Comment (BaurzhanSakhariev) [reply server/src/testFixtures/java/io/crate/testing/QueryTester.java:92]: Sorry for pushing in the wrong direction (I meant using custom `LongSupplier` impl but asserting oids everywhere is indeed not readable)

Basically https://github.com/crate/crate/commit/99e697e00e84bfc8a34ad1a501bbc686dc07dc91 was okay in that regard that it was a dedicated test, with simple custom supplier to kick new logic the only adjustment would be to follow 
> Similar to test_is_null_on_columns_without_doc_values (minus the _field_names stuff)
and assert not `queryTester.toQuery` output but `queryTester.runQuery` output -> which would be failing without the fix even with custom "return constant" supplier.

Comment (jeeminso) [reply server/src/testFixtures/java/io/crate/testing/QueryTester.java]: Thank you both for looking into this. Could I make just one more [attempt](https://github.com/crate/crate/pull/14908/commits/f819a98e584092010831ac4d95740bb6550effb2) hoping the existing tests account for oids.

Comment (BaurzhanSakhariev) [reply server/src/testFixtures/java/io/crate/testing/QueryTester.java]: I have 2 questions
1.  > and actually found another missing conversion from NumTermsPerDocQuery.

Is this covered somewhere in the test? I see change in relevant `testNeqAnyOnArrayLiteral()` but test_is_null_on_columns_without_doc_values seems to be irrelevant - I feel like problems in numTermsPerDoc should be somewhere in `ArrayLengthQueryBuilderTest`

2. Current approach with checking query and keep it human readable via `queryTester.columnNameToOid` looks fine to me but let's wait for other opinions (cc @seut, @mfussenegger). I think another approach is to not assert query at all but instead check output of `queryTester.run()`?

Comment (jeeminso) [reply server/src/testFixtures/java/io/crate/testing/QueryTester.java]: https://github.com/crate/crate/pull/14908/commits/f819a98e584092010831ac4d95740bb6550effb2 allows lucene query tests to use oids directly or `queryTester.columnNameToOid("x")` to make it more readable.

1. The changes to `test_is_null_on_columns_without_doc_values` was just to describe how existing query tests will be updated.
2. > I think another approach is to not assert query at all but instead check output of queryTester.run()

I think `runQuery` is slightly more time consuming since we need to index and execute, we could also wait for others opinions on this.
3. https://github.com/crate/crate/commit/3acc06c5973f795328f450e3455fe1d0f7b783f5 will be updated to use `queryTester.columnNameToOid`, so I am hoping all existing query tests will be updated accordingly.

**Review by BaurzhanSakhariev — APPROVED** [review]
LGTM, please wait for at least one more approval.

Thanks for catching!

File: server/src/testFixtures/java/io/crate/testing/QueryTester.java
Comment (seut) [inline server/src/testFixtures/java/io/crate/testing/QueryTester.java:96]: Not sure, but checkstyle may complain.
```suggestion
                () -> COLUMN_OID_UNASSIGNED,
```

**Review by seut — APPROVED** [review]
👍 thx

**Pull request merged**
